src/main/java/com/epam/esm/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.dao.CertificateDAO;
+import com.epam.esm.dao.OrderDAO;
+import com.epam.esm.dao.UserDAO;
+import com.epam.esm.model.dto.OrderDTO;
+import com.epam.esm.model.entity.Certificate;
+import com.epam.esm.model.entity.Order;
+import com.epam.esm.model.entity.Page;
+import com.epam.esm.model.entity.User;
+import com.epam.esm.model.exception.CertificateNotFoundException;
+import com.epam.esm.model.exception.OrderNotFoundException;
+import com.epam.esm.model.exception.UserNotFoundException;
+import com.epam.esm.service.OrderService;
+import com.epam.esm.util.MapperDTO;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderDAO orderDAO;
+    private final CertificateDAO certificateDAO;
+    private final UserDAO userDAO;
+    private final MapperDTO mapperDTO;
+
+
+    @Override
+    public List<OrderDTO> findAll(Page page) {
+        return orderDAO.findAll(page)
+            .stream()
+            .map(mapperDTO::convertOrderToDTO)
+            .collect(Collectors.toList());
+    }
+
+    @Override
+    public OrderDTO findById(Long id) {
+        return mapperDTO.convertOrderToDTO(orderDAO.findById(id)
+            .orElseThrow(() -> new OrderNotFoundException(id.toString())));
+    }
+
+    @Override
+    @Transactional
+    public OrderDTO create(OrderDTO orderDTO) {
+        orderDTO.setCost(new BigDecimal(0));
+        List<Certificate> certificates = new ArrayList<>();
+        Order order = mapperDTO.convertDTOToOrder(orderDTO);
+        User user = userDAO.findById(orderDTO.getUserId())
+            .orElseThrow(() -> new UserNotFoundException(orderDTO.getUserId().toString()));
+        order.setUser(user);
+        orderDTO.getCertificateId().forEach(id -> {
```

Comment:
too big peace of code, split the method. add getCertificates method to retrieve order certificates from DB